lineman/app/services/thread_query_service.coffee

```diff
@@ -1,40 +1,37 @@
 angular.module('loomioApp').factory 'ThreadQueryService', (Records) ->
   new class ThreadQueryService
 
-    filterQuery: (filter) ->
-      threadQueryFor createBaseView(), filter
-
-    groupQuery: (group = {}, options = {}) ->
-      threadQueryFor createGroupView(group), 
-                     options['filter'] or 'show_unread',
-                     options['limit'] or 5
+    filterQuery: (filter, options = {}) ->
+      threadQueryFor createBaseView(),
+                     filter,
+                     options['queryType'] or 'all'
 
     timeframeQuery: (options = {}) ->
       threadQueryFor createTimeframeView(options['name'], options['timeframe']),
                      options['filter'] or 'show_all',
-                     options['limit'] or 100
+                     options['queryType'] or 'timeframe'
+
+    groupQuery: (group = {}, options = {}) ->
+      threadQueryFor createGroupView(group),
+                     options['filter'] or 'show_unread',
+                     options['queryType'] or 'inbox'
 
-    threadQueryFor = (view, filter) ->
-      view: view
-      filter: createFilter(filter)
-      threads: ->
-        @filter @view.data()
-      length: ->
-        @threads().length
-      any: ->
-        @length() > 0
+    threadQueryFor = (view, filters, queryType) ->
+      view:       view
+      filter:     createFilter([].concat(filters), queryType)
+      threads: -> @filter @view.data()
```

Comment:
how does this work? -> will give you the wrong this/@. I would have expected you needed to use => here